specification/containerstorage/ContainerStorage.Management/main.tsp

```diff
@@ -14,12 +14,18 @@ using Azure.ResourceManager.Foundations;
 @armProviderNamespace
 @service({
   title: "Azure-ContainerStorageRP",
-  version: "2023-07-01-preview",
 })
+@versioned(Versions)
 @doc("Azure Container Storage Resource Provider API.")
-@useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
 namespace Microsoft.ContainerStorage;
 
+/** Api versions */
+enum Versions {
+  /** 2023-07-01-preview preview version */
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
```

Comment:
Should have ARM common type version reference.

Reply:
This isn't blocking for this PR so I won't do it here. THere is a lot of spec not respecting that. We can have a dedicated one that update. Just trying ot merge this before the linter rules start failing again